### PR TITLE
Randomness

### PR DESCRIPTION
**Pull request by hyd628**
### Description/Original PRs

https://github.com/PureStake/moonbeam-docs/pull/377

### Checklist

- [x] If this requires removing old images from the `moonbeam-docs` repo, I have created a corresponding PR
- [x] If this requires adding/updating/deleting redirects, I have created a corresponding PR in the `moonbeam-mkdocs` repo
- [x] If this requires removing old variables from the `moonbeam-docs` repo, I have created a corresponding PR

### Corresponding PRs

Please link to any additional corresponding PRs here.

### CODE REVIEW

File: builders/pallets-precompiles/precompiles/randomness.md [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+
+一般而言，彩票合约包含查看随机数请求状态的功能，其将被用于决定彩票是否将持续接受新的参与者、是否开始或是否过期。它将会根据您希望使用的随机数来源使用`requestLocalVRFRandomWords`或`requestRelayBabeEpochRandomWords`函数选取随机词。除外，它将会执行`fulfillRandomWords`方法并回调将会完成请求以及使用随机词随机地选取彩票获胜者。
+
+同样合约中具有一些能够编辑的常量，尤其是能够用于产生独特结果的`SALT_PREFIX` 。
+
+### Remix设置 {: #remix-set-up }
+
+您可以使用[Remix](https://remix.ethereum.org/){target=_blank}与随机数预编译和消费者合约交互。要将接口新增至Remix并跟随其后教程，您需要跟随以下步骤：
+
+1. 复制[`RandomnessLotteryDemo.sol`](https://github.com/PureStake/moonbeam/blob/4e2a5785424be6faa01cd14e90155d9d2ec734ee/tests/contracts/solidity/RandomnessLotteryDemo.sol){target=_blank}文件
```
Comment (eshaben) [inline builders/pallets-precompiles/precompiles/randomness.md:154]:
```suggestion
1. 复制[`RandomnessLotteryDemo.sol`](https://raw.githubusercontent.com/PureStake/moonbeam-docs/master/.snippets/code/randomness/RandomnessLotteryDemo.sol){target=_blank}文件
```

File: builders/pallets-precompiles/precompiles/randomness.md [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+如您使用预设合约，您需要准备以下内容：
+
+- [安装成功的MetaMask并连接至Moonbase Alpha](/tokens/connect/metamask/){target=_blank}
+- 在Moonbase Alpha上创建或是拥有3个账户以测试彩票合约
+- 所有账户皆需拥有`DEV ` Token
+ --8<-- 'text/faucet/faucet-list-item.md'
+
+### 范例彩票合约 {: #example-contract }
+
+在此教程中，您将会与使用随机数预编译和消费者接口的彩票合约交互。您将会生成用于公平选取彩票获胜者的随机词。您可以在Moonbeam Docs GitHub库中寻找将被用于此教程的彩票合约文档，[`RandomnessLotteryDemo.sol`](https://raw.githubusercontent.com/PureStake/moonbeam-docs/blob/master/.snippets/code/randomness/RandomnessLotteryDemo.sol){target=_blank}。
```
Comment (eshaben) [inline builders/pallets-precompiles/precompiles/randomness.md:142]:
```suggestion
在此教程中，您将会与使用随机数预编译和消费者接口的彩票合约交互。您将会生成用于公平选取彩票获胜者的随机词。您可以在Moonbeam Docs GitHub库中寻找将被用于此教程的彩票合约文档，[`RandomnessLotteryDemo.sol`](https://raw.githubusercontent.com/PureStake/moonbeam-docs/master/.snippets/code/randomness/RandomnessLotteryDemo.sol){target=_blank}。
```

**Review by eshaben — COMMENTED** [review]
thanks! just one thing to double check on please!

File: builders/pallets-precompiles/precompiles/randomness.md [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+
+## 请求和完成过程 {: #request-and-fulfill-process }
+
+要启用随机性，您必须有一个执行以下操作的合约：
+
+  - 导入`Randomness.sol`预编译和`RandomnessConsumer.sol`接口
+  - 从`RandomnessConsumer.sol`接口继承
+  - 根据您希望使用的随机数资源，通过预编译的[`requestLocalVRFRandomWords`方法](#:~:text=requestLocalVRFRandomWords)或[`requestRelayBabeEpochRandomWords`方法](#:~:text=requestRelayBabeEpochRandomWords)请求随机数
+  - 通过预编译的[`fulfillRequest`方法](#:~:text=fulfillRequest)完成请求
+  - 使用与[`fulfillRandomWords`方法相同的签名](#:~:text=fulfillRandomWords(uint256 requestId, uint256[] memory randomWords))通过`fulfillRandomWords`方法使用随机数
```
[the remark this answers is not in the document]
Comment (hyd628) [reply builders/pallets-precompiles/precompiles/randomness.md]: Fixed. Thanks

**Comment by hyd628** [comment]
Thanks @eshaben , incorporated the feedback.

File: builders/pallets-precompiles/precompiles/randomness.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (eshaben) [reply builders/pallets-precompiles/precompiles/randomness.md]: Thank you!

**Review by eshaben — APPROVED** [review]
LGTM ✨

**Pull request merged**